### PR TITLE
Kafka integration test

### PR DESCRIPTION
**Pull request by robbavey**
## What does this PR do?

This PR adds integration tests for the Kafka output, with the use cases, and some of the mechanics around reading events from a kafka topic ported from the beats output. 

This PR also includes infrastructure to create a test docker image with Kafka, exposing ports to allow testing over plaintext and SSL.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
~- [ ] I have added an entry in `CHANGELOG.md` or `CHANGELOG-developer.md`.~

## How to test this PR locally

- Run `docker-compose up` in the `testing/environment/docker` folder
- Run `mage integrationTest`

Note that I needed to delete the `docker_kafka` container after each test run to allow the tests to continue to run.

## Related issues

Relates #166, #176

### CODE REVIEW

**Review by faec — APPROVED** [review]
This looks great, thank you. It looks like there will be some followup to make these tests runnable in CI / without manually starting containers, but considering the impending holiday I'm happy to worry about that after the other pending integration tests are merged.

File: testing/environment/docker/docker-compose.yml
Comment (faec) [inline testing/environment/docker/docker-compose.yml:4]: Oh, missed this initially, it looks like this file needs to have the home directory stripped out or maybe just be deleted if you were just using it for testing the prototype

**Comment by faec** [comment]
One other thing if you have time before merging: can you put this in `testing/environments/kafka` instead of `testing/environment/docker`? (My pending ES container is in `testing/environments/elasticsearch`)

File: testing/environment/docker/docker-compose.yml
Comment (robbavey) [reply testing/environment/docker/docker-compose.yml]: 🤦‍♂️  Done

**Comment by robbavey** [comment]
I've moved the files @faec

**Comment by faec** [comment]
Looks great, thank you again! Don't worry about waiting on @fearful-symmetry for merge, they are already out for the holiday :-)

**Pull request merged**
